### PR TITLE
chore: Session Replay Raw Usage Estimation Effort - NEWRELIC-8153

### PR DESCRIPTION
**Pull request by metal-messiah**
DO NOT MERGE testing rrweb settings
---
### Overview
This PR represents a TEMPORARY effort to evaluate raw usage of Session Replay.  Observation of `bytes`, `nodes`, `initial snapshot bytes`, `compressed initial snapshot bytes`, and `bytesPerMinute` will be attempted for each of "standard" and "spa" type web applications.

### Related Issue(s)
NEWRELIC-8153

### CODE REVIEW

File: src/features/metrics/aggregate/session-replay-investigation.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    if (metrics.Nodes === 2) {
+      gzip(strToU8(jsonStr), (err, data) => {
+        if (err) return
+        metrics.InitialSnapshotBytesCompressed = data.length
+      })
+      metrics.InitialSnapshotBytes += jsonStr.length // compressed string?
+    }
+    metrics.Bytes += bytes
+    metrics.BytesPerMinute = Math.round(metrics.Bytes / performance.now() * 60000)
+    metrics.NodesPerMinute = Math.round(metrics.Nodes / performance.now() * 60000)
```
Comment (bjfield) [inline src/features/metrics/aggregate/session-replay-investigation.js:33]: I'm not sure why, but `NodesPerMinute` is the one metric that appears not to be present in the sm payload. All the others are there for both spa and standard.

File: src/features/metrics/aggregate/index.js [hunk 2]
```diff
@@ -126,6 +128,24 @@ export class Aggregate extends AggregateBase {
...
+        // PageSession/Feature/SessionReplay/Spa/BytesPerMinute
+        // PageSession/Feature/SessionReplay/Spa/NodesPerMinute
+        // ---- or ----
+        // PageSession/Feature/SessionReplay/Standard/Nodes
+        // PageSession/Feature/SessionReplay/Standard/Bytes
+        // PageSession/Feature/SessionReplay/Standard/InitialSnapshotBytes
+        // PageSession/Feature/SessionReplay/Standard/InitialSnapshotBytesCompressed
+        // PageSession/Feature/SessionReplay/Standard/BytesPerMinute
+        // PageSession/Feature/SessionReplay/Standard/NodesPerMinute
+        if (val > 0) this.storeSupportabilityMetrics(`PageSession/Feature/SessionReplay/${key}/${type}`, val)
```
[the remark this answers is not in the document]
Comment (metal-messiah) [reply src/features/metrics/aggregate/index.js:147]: I think the SM string is actually the better way to format.  I will update the comments

File: src/features/metrics/aggregate/session-replay-investigation.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+export let type = 'Standard'
+window.addEventListener('popstate', function () {
+  type = 'Spa'
+})
+
+const stopRecording = record({
+  emit: (event) => {
+    metrics.Nodes++
+    const jsonStr = stringify(event)
```
[the remark this answers is not in the document]
Comment (metal-messiah) [reply src/features/metrics/aggregate/session-replay-investigation.js:22]: Depends what very large means, but we've seen some payloads of upwards of 2-3MB.  Not sure what the performance hit is on that

File: src/features/metrics/aggregate/session-replay-investigation.js (hunk 1 above)
Comment (metal-messiah) [reply src/features/metrics/aggregate/session-replay-investigation.js:33]: https://chart-embed.staging-service.newrelic.com/herald/fba6feff-1c98-4bef-92c8-9ae31ed227d2?height=400px&timepicker=true

File: src/features/metrics/aggregate/session-replay-investigation.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrickhousley) [reply src/features/metrics/aggregate/session-replay-investigation.js:23]: We saw the stringify on those 2-3MB payloads run really fast. I don't know if we know what too large is for stringify.

**Review by patrickhousley — APPROVED** [review]

**Review by bjfield — APPROVED** [review]
Sends all the specified metrics for both spa and standard, although I haven't been able to verify MaxTimeHit.

**Comment by metal-messiah** [comment]
⚠️ This PR needs security and privacy approval before merging

**Comment by metal-messiah** [comment]
Security approval has been met. Still waiting on Privacy/Legal approval

**Comment by jomle** [comment]
Tweaking some rrweb settings so I can get a PR build to test with. DO NOT MERGE

**Pull request closed without merging**